src/main/java/net/fabricmc/mappingio/format/MappingFormat.java

```diff
@@ -188,4 +295,517 @@ public String getGlobPattern() {
+			}
+
+			public OptionalFeature srcNames() {
+				return names.srcNames;
+			}
+
+			public OptionalFeature dstNames() {
+				return names.dstNames;
+			}
+
+			public OptionalFeature srcDescs() {
+				return descriptors.srcDescriptors;
+			}
+
+			public OptionalFeature dstDescs() {
+				return descriptors.dstDescriptors;
+			}
+
+			private OptionalFeature positions;
+			private OptionalFeature lvIndices;
+			private OptionalFeature lvtRowIndices;
+			private OptionalFeature startOpIndices;
+			private OptionalFeature endOpIndices;
+			private NameFeature names;
+			private DescFeature descriptors;
+		}
+
+		public static class NameFeature {
+			NameFeature() {
+				this(false);
+			}
+
+			NameFeature(boolean initWithFullSupport) {
+				this(initWithFullSupport ? OptionalFeature.OPTIONAL : OptionalFeature.UNSUPPORTED,
+						initWithFullSupport ? OptionalFeature.OPTIONAL : OptionalFeature.UNSUPPORTED);
+			}
+
+			private NameFeature(OptionalFeature srcNames, OptionalFeature dstNames) {
+				this.srcNames = srcNames;
+				this.dstNames = dstNames;
+			}
+
+			public NameFeature withSrcNames(OptionalFeature srcNameFeature) {
+				this.srcNames = srcNameFeature;
+				return this;
+			}
+
+			public NameFeature withDstNames(OptionalFeature dstNameFeature) {
+				this.dstNames = dstNameFeature;
+				return this;
+			}
+
+			@Override
+			public NameFeature clone() {
+				return new NameFeature(srcNames, dstNames);
+			}
+
+			public OptionalFeature srcNames() {
+				return srcNames;
+			}
+
+			public OptionalFeature dstNames() {
+				return dstNames;
+			}
+
+			private OptionalFeature srcNames;
+			private OptionalFeature dstNames;
+		}
+
+		public static class DescFeature {
+			DescFeature() {
+				this(false);
+			}
+
+			DescFeature(boolean initWithFullSupport) {
+				this(initWithFullSupport ? OptionalFeature.OPTIONAL : OptionalFeature.UNSUPPORTED,
+						initWithFullSupport ? OptionalFeature.OPTIONAL : OptionalFeature.UNSUPPORTED);
+			}
+
+			private DescFeature(OptionalFeature srcDescriptors, OptionalFeature dstDescriptors) {
+				this.srcDescriptors = srcDescriptors;
+				this.dstDescriptors = dstDescriptors;
+			}
+
+			public DescFeature withSrcDescs(OptionalFeature srcDescriptorFeature) {
+				this.srcDescriptors = srcDescriptorFeature;
+				return this;
+			}
+
+			public DescFeature withDstDescs(OptionalFeature dstDescriptorFeature) {
+				this.dstDescriptors = dstDescriptorFeature;
+				return this;
+			}
+
+			@Override
+			public DescFeature clone() {
+				return new DescFeature(srcDescriptors, dstDescriptors);
+			}
+
+			public OptionalFeature srcDescs() {
+				return srcDescriptors;
+			}
+
+			public OptionalFeature dstDescs() {
+				return dstDescriptors;
+			}
+
+			private OptionalFeature srcDescriptors;
+			private OptionalFeature dstDescriptors;
+		}
+
+		public enum OptionalFeature {
+			REQUIRED,
+			OPTIONAL,
+			UNSUPPORTED
+		}
+
+		public enum ElementCommentSupport {
+			NAMESPACED,
+			SHARED,
```

Comment:
What's the purpose here if we don't have a namespace-accepting visitComment method in the visitor?

Reply:
Mostly for future-proofing, the namespaced `visitComment` is going to get added in a future version